packages/cozy-konnector-libs/src/libs/saveIdentity.js

```diff
@@ -75,12 +78,49 @@ const saveIdentity = async (
     identity.contact.address = formatAddress(identity.contact.address)
   }
 
-  await updateOrCreate(
-    [identity],
-    'io.cozy.identities',
-    ['identifier', 'cozyMetadata.createdByApp'],
-    { ...options, sourceAccountIdentifier: accountIdentifier }
-  )
+  if (options.merge == false) {
+    // Using cozy client in a none merging strategy here.
+    const newClient = cozyClient.new
+    const { data: existingIdentity } = await newClient.query(
+      Q('io.cozy.identities')
+        .where({
+          cozyMetadata: { createdByApp: manifest.data.slug },
+          identifier: accountIdentifier
+        })
+        .indexFields(['cozyMetadata.createdByApp'])
```

Comment:
You should have a warning in your console with this request, saying that `identifier` should be in the `indexFields`. No? 

This should be something like: 
```
Q('io.cozy.identities')
        .where({
          cozyMetadata: { createdByApp: manifest.data.slug },
          identifier: accountIdentifier
        })
        .indexFields(['identifier', 'cozyMetadata.createdByApp'])
```

Reply:
No i haven't

Reply:
Which cozy-client's version do you use?!

Reply:
But I found this message "passing a io.cozy.contacts object is deprected, please pass a full identity object"

Reply:
I use "cozy-client": already "^29.1.2" import by the connectors-libs

Reply:
So old... You should definitely upgrade it.

Reply:
I upgraded to the latest "cozy-client": "^33.2.0"

But i have some side effect, on a test in BaseKonnector.js

Summary of all failing tests
 FAIL  packages/cozy-konnector-libs/src/libs/BaseKonnector.spec.js
  ● Test suite failed to run

    TypeError: The "original" argument must be of type function. Received undefined

      at Object.<anonymous> (node_modules/open/index.js:9:17)
      at Object.<anonymous> (node_modules/cozy-client/dist/cli/index.js:18:36)